magnet_coils.py

```diff
@@ -0,0 +1,247 @@
+            must be either a circle or rectangle. For a circular cross-section,
+            the list format is
+            ['circle' (str), radius (float, cm)]
+            For a rectangular cross-section, the list format is
+            ['rectangle' (str), width (float, cm), thickness (float, cm)]
+        start (int): index for line in data file where coil data begins.
+        stop (int): index for line in data file where coil data ends (defaults
+            to None).
+        name (str): name used for exported STEP file.
+    """
+    # Extract coil cross-section shape
+    shape = cross_section[0]
+    # Conditionally extract parameters for circular cross-section
+    if shape == 'circle':
+        if len(cross_section) != 2:
+            raise ValueError(
+                'Format of list defining circular cross-section must be\n'
+                '[\'circle\' (str), radius (float, cm)]'
+            )
+        radius = cross_section[1]
+    # Conditinally extract parameters for rectangular cross-section
+    elif shape == 'rectangle':
+        if len(cross_section) != 3:
+            raise ValueError(
+                'Format of list defining rectangular cross-section must be\n'
+                '[\'rectangle\' (str), width (float, cm), thickness '
+                '(float, cm)]'
+            )
+        width = cross_section[1]
+        thickness = cross_section[2]
+    else:
+        raise ValueError(
+            'Magnet cross-section must be either a circle or rectangle. The '
+            'first entry of the list defining the cross-section must be the '
+            'shape, with the following entries defining the shape parameters.\n'
+            '\n'
+            'For a circular cross-section, the list format is\n'
+            '[\'circle\' (str), radius (float, cm)]\n'
+            '\n'
+            'For a rectangular cross-section, the list format is\n'
+            '[\'rectangle\' (str), width (float, cm), thickness (float, cm)]'
+        )
+    
+    # Load magnet coil data
+    file_obj = open(file, 'r')
+    data = file_obj.readlines()[start:stop]
+
+    # Initialize Cubit
+    cubit.init([''])
+    cubit.cmd('undo on')
+
+    # Initialize vertex, curve, and surface indices
+    vert_id = 0
+    curve_id = 0
+    surf_id = 0
+
+    # Initialize path list
+    path = []
+
+    # Extract magnet coil data
+    for line in data:
+        # Parse line in magnet coil data
+        line = line.strip()
+        columns = line.split()
+
+        # Break at indicator signaling end of data
+        if columns[0] == 'end':
+            break
+
+        # Extract coordinates for vertex in path
+        x = float(columns[0])*100
+        y = float(columns[1])*100
+        z = float(columns[2])*100
+        # Extract coil current
+        s = float(columns[3])
+        
+        # Create vertex in path
+        cubit.cmd('create vertex ' + str(x) + ' ' + str(y) + ' ' + str(z))
```

Comment:
f-strings are a very convenient and more compact alternative to concatenation: https://docs.python.org/3/tutorial/inputoutput.html

Reply:
This is really cool, thanks!